packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/DiagramServices.java

```diff
@@ -109,4 +109,13 @@ public boolean isFaded(IDiagramElement diagramElement) {
         }
         return isFaded;
     }
+
+    @Override
+    public boolean isDiagramEmpty(IDiagramService diagramService) {
```

Comment:
This must be new since the last time I reviewed this PR because this cannot work. I could accept that in the services of Papaya diagrams because it would work in this case but imagine that I have 1 node in my diagram which has 10 child nodes and the view deletion request is for one of its child node then this code does not work.

Imagine now that I have 3 nodes in my diagram and a tool triggers the creation of 3 view deletion request then this code would not work either. Finding out that a diagram is empty in all use case is more complex than what you algorithm suggest. Please move both the implementation of this service and its definition out of IDiagramServices for now and into something in the Java services of Papaya (since I suspect that you are only using it there).

You could also improve this algorithm to make it work all the time but it will require some testing of corner cases. Consumers of Sirius Web such as SysON would be quite interested by such service.